tool/gravity/cli/config.go

```diff
@@ -736,19 +726,14 @@ func (j *JoinConfig) GetPeers() ([]string, error) {
 }
 
 // GetRuntimeConfig returns the RPC agent runtime configuration
-func (j *JoinConfig) GetRuntimeConfig() (*proto.RuntimeConfig, error) {
-	config := &proto.RuntimeConfig{
+func (j *JoinConfig) GetRuntimeConfig() proto.RuntimeConfig {
+	return proto.RuntimeConfig{
 		Token:        j.Token,
 		Role:         j.Role,
 		SystemDevice: j.SystemDevice,
 		DockerDevice: j.DockerDevice,
 		Mounts:       convertMounts(j.Mounts),
 	}
-	err := install.FetchCloudMetadata(j.CloudProvider, config)
```

Comment:
Why is this being removed?

Reply:
I found that this is already being done by the `install.NewAgent()` function which join peer calls so it was extra work. Moreover, cloud provider is actually unknown at this particular point b/c join peer retrieves it from the cluster now (which is done after connecting to wizard or cluster inside the peer).